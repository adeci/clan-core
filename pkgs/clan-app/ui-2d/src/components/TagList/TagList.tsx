import { Component, For } from "solid-js";
import { Typography } from "@/src/components/Typography";
import "./TagList.css";

interface TagListProps {
  values: string[];
}

export const TagList: Component<TagListProps> = (props) => {
  return (
    <div class="tag-list">
      <For each={props.values}>
        {(tag) => (
          <Typography hierarchy="label" size="s" inverted={true} class="tag">
            {tag}
          </Typography>
        )}
      </For>
    </div>
  );
};
