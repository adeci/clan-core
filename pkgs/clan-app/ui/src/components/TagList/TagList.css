div.tag-list {
  @apply flex flex-wrap gap-2;

  span.tag {
    @apply w-fit rounded-full px-3 py-2 bg-inv-4 fg-inv-1;
  }
}
