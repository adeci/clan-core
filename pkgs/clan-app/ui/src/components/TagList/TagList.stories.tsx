import type { Meta, StoryObj } from "@kachurun/storybook-solid";
import { TagList, TagListProps } from "./TagList";

const meta: Meta<TagListProps> = {
  title: "Components/TagList",
  component: TagList,
  decorators: [
    // wrap in a fixed width div so we can check that it wraps
    (Story) => {
      return (
        <div style={{ width: "20em" }}>
          <Story />
        </div>
      );
    },
  ],
};

export default meta;

type Story = StoryObj<TagListProps>;

export const Default: Story = {
  args: {
    values: [
      "Titan",
      "Enceladus",
      "Mimas",
      "Dione",
      "Iapetus",
      "Tethys",
      "Hyperion",
      "Epimetheus",
    ],
  },
};
