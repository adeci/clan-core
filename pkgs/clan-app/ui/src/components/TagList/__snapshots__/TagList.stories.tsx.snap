// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/TagList Default smoke-test 1`] = `
<div style="width: 20em;">
  <div class="tag-list">
    <span class="fnt-clr-primary fnt-clr--inverted fnt-label-s fnt-weight-normal tag">
      Titan
    </span>
    <span class="fnt-clr-primary fnt-clr--inverted fnt-label-s fnt-weight-normal tag">
      Enceladus
    </span>
    <span class="fnt-clr-primary fnt-clr--inverted fnt-label-s fnt-weight-normal tag">
      Mimas
    </span>
    <span class="fnt-clr-primary fnt-clr--inverted fnt-label-s fnt-weight-normal tag">
      Dione
    </span>
    <span class="fnt-clr-primary fnt-clr--inverted fnt-label-s fnt-weight-normal tag">
      Iapetus
    </span>
    <span class="fnt-clr-primary fnt-clr--inverted fnt-label-s fnt-weight-normal tag">
      Tethys
    </span>
    <span class="fnt-clr-primary fnt-clr--inverted fnt-label-s fnt-weight-normal tag">
      Hyperion
    </span>
    <span class="fnt-clr-primary fnt-clr--inverted fnt-label-s fnt-weight-normal tag">
      Epimetheus
    </span>
  </div>
</div>
`;
